### ISSUE TITLE
Redux typings

### ISSUE THREAD

**Issue by ulfryk** [reactjs/redux#1401]
Current typings on *DefinitelyTyped* seem to be outdated and not so strict. 
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ffceea9dd124d277c4597c7bd12930666ec074c5/redux/redux.d.ts

```
// Type definitions for Redux v1.0.0
// Project: https://github.com/rackt/redux
```

I have prepared up-to-date version with more strict approach to type safety -> [see it here](https://gist.github.com/ulfryk/69981ccfb488647b6ee3).

Please update it or add typings to this repo. Or I can update it myself. I can help updating those typings also on future releases :)

**Comment by gaearon** [comment]
Please send a PR to DefinitelyTyped. We don’t maintain them ourselves but your help is very welcome.

**Comment by aikoven** [comment]
@ulfryk As you could see, it's been a while since https://github.com/DefinitelyTyped/DefinitelyTyped/pull/7908 was submitted, and its predecessor https://github.com/DefinitelyTyped/DefinitelyTyped/pull/6629 was stuck for even longer time without approval.

It seems that DefinitelyTyped model outgrown itself and we should move to [Typings](https://github.com/typings/typings).

**Comment by ulfryk** [comment]
@aikoven - yes I see - my proposition is also on it's way to stuck ;) ( [8093](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8093)

**Comment by wbuchwalter** [comment]
Another option would be to maintain them directly in the repo, like what is done for [ng-redux](https://github.com/wbuchwalter/ng-redux), with an `index.d.ts` at the root.
This seems to be the best solution since TypeScript 1.6, see [Typings for npm packages](https://github.com/Microsoft/TypeScript/wiki/Typings-for-npm-packages), and doesn't require any action from the user to get the typings working.
Your call @gaearon, I can understand your wish of keeping redux free of specific languages considerations as well.

**Comment by gaearon** [comment]
I’m happy to keep it in our repo if somebody is willing to maintain them.

**Comment by wbuchwalter** [comment]
:+1: 
We have been ~10 recurrent people debating on this issue in the various PR at definitlTtyped, so I think there will be plenty of people willing to help on that, myself included.
Plus, if you make a change to Redux's API, outdated typings will only affect TypeScript user, so it will force the community to act, and have no impact on non-TS users.

@aikoven, are you ok to bring back your PR here as something compliant with [TS 1.6 definition files for npm modules](https://github.com/Microsoft/TypeScript/wiki/Typings-for-npm-packages) ?

Does anyone see any problem with this approach that I may have missed?

**Comment by ulfryk** [comment]
:+1:  I can help maintaining TS typings in this repo. That is definitely the most user friendly way :)

**Comment by aikoven** [comment]
@wbuchwalter I'll do it.

**Comment by abarre** [comment]
I think that the approach of keeping the definition file in this repo is better.
So, +1.

**Comment by xogeny** [comment]
+1

Keeping a typing a here is much better in my opinion. I've had very good luck with npm+TS using latest versions of TS.

**Comment by aikoven** [comment]
Done.

**Comment by gaearon** [comment]
Please help review the new official typings in https://github.com/reactjs/redux/pull/1413.

**Comment by gaearon** [comment]
Typing from #1431 have been released with Redux 3.4.0. Big thanks to everyone who made this happen!
